### PR TITLE
feat(macos): allow to disable system tray menu on left click, closes #317

### LINKED ISSUE #317: [bug] SystemTray shows the menu on `LeftClick`, it should be on `RightClick` only
### Describe the bug

default SystemTrayEvent::LeftClick will show Tray Menu, I not found API or setting to  disable the behavior.

https://tauri.studio/docs/guides/system-tray

```rust
use tauri::{CustomMenuItem, SystemTray, SystemTrayMenu};
use tauri::Manager;

fn main() {
  let tray_menu = SystemTrayMenu::new(); // insert the menu items here
  tauri::Builder::default()
    .system_tray(SystemTray::new().with_menu(tray_menu))
    .on_system_tray_event(|app, event| match event {
      SystemTrayEvent::LeftClick {
        position: _,
        size: _,
        ..
      } => {
        // the will auto pop Tray Menu
        println!("system tray received a left click");
      }
      SystemTrayEvent::RightClick {
        position: _,
        size: _,
        ..
      } => {
        println!("system tray received a right click");
      }
      SystemTrayEvent::DoubleClick {
        position: _,
        size: _,
        ..
      } => {
        println!("system tray received a double click");
      }
      SystemTrayEvent::MenuItemClick { id, .. } => {
        match id.as_str() {
          "quit" => {
            std::process::exit(0);
          }
          "hide" => {
            let window = app.get_window("main").unwrap();
            window.hide().unwrap();
          }
          _ => {}
        }
      }
      _ => {}
    })
    .run(tauri::generate_context!())
    .expect("error while running tauri application");
}
```

### Reproduction

_No response_

### Expected behavior

_No response_

### Platform and versions

```shell
Operating System - Mac OS, version 12.2.1 X64

Node.js environment
  Node.js - 16.13.2
  @tauri-apps/cli - 1.0.0-rc.2
  @tauri-apps/api - 1.0.0-rc.0

Global packages
  npm - 8.1.2
  pnpm - 6.30.0
  yarn - 1.22.17

Rust environment
  rustc - 1.58.1
  cargo - 1.58.0

Rust environment
  rustup - 1.24.3
  rustc - 1.58.1
  cargo - 1.58.0
  toolchain - stable-x86_64-apple-darwin 

App directory structure
/dist
/node_modules
/src-tauri
/src

App
  tauri.rs - 1.0.0-rc.0
  build-type - bundle
  CSP - default-src 'self'
  distDir - ../dist
  devPath - http://localhost:3000/
  framework - React
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by wusyong** [comment]
I think I mentioned that default tray supports both clicks too.
If we really want to specify certain behaviour, I guess we expose a method for extension trait.

**Comment by amrbashir** [comment]
A lot of users asked for this behaviour, check the linked issue and this one https://github.com/tauri-apps/tauri/issues/4002 and some on discord too.

**Comment by wusyong** [comment]
I know, but as I stated in https://github.com/tauri-apps/tao/issues/317 even official trays works on both or even just left click.
Supporting both clicks is the sensible default. User won't get confused on first try.
If we really want to specify only one kind of click, it should be a extension trait method IMHO.

**Comment by amrbashir** [comment]
Alright, a trait extension sounds good.

**Comment by amrbashir** [comment]
I attempted to make it as a trait extension but becaue of my lack of macOS development skill, I couldn't finish it. @wusyong whenever you have some time, could u look into this? I added a `TODO` where the value should be read.

File: src/platform/macos.rs [hunk 1]
```diff
@@ -515,7 +515,7 @@ impl SystemTrayExtMacOS for SystemTray {
     self.0.icon_is_template = is_template
   }
 
-  fn set_icon_as_template(&mut self, enable: bool) {
+  fn set_menu_on_left_click(&mut self, enable: bool) {
```
Comment (wusyong) [inline src/platform/macos.rs:518]: It's good to me but please default with left click. Official tray icons are default with both clicks or left click.
None of them is right click only.

Comment (amrbashir) [reply src/platform/macos.rs:518]: yeah that's the default, there was a typo, fixed in ed34d29

**Pull request merged**